### ISSUE TITLE
How about the performance of CatK in larger model?

### ISSUE THREAD

**Issue by JasonZhou404** [NVlabs/catk#3]
The work is based on the smart-tiny, did you guys do any expertiment on larger model, like smart-large, would it also improves the metrics?

**Comment by zhejz** [comment]
No we haven’t. The 7M-model is the biggest model we trained, and it outperforms a 3M model. Be aware that training SMART-tiny-7M on 8x A100 takes already 5 days.

**Comment by OrangeSodahub** [comment]
@zhejz I found that the data preprocess script in catK is much different from in SMART, especially for the map data. So are the `pkl` files generated by your script also compatible for the training script of original SMART? If not, what advantages does your processing script have compared to SMART?

**Comment by zhejz** [comment]
We only generated the pickle file for the agent action tokens ([here](https://github.com/NVlabs/catk/blob/main/src/smart/tokens/agent_vocab_555_s2.pkl)), not for the map tokens. The [map token vocab pickle file](https://github.com/NVlabs/catk/blob/main/src/smart/tokens/map_traj_token5.pkl) is exactly the same as the one in SMART's repository.

Our data preprocess script adds more map information (e.g., stop signs), some tricks we found to be useful (e.g., replace the crosswalk/speed_bump polygon with polylines), and includes some data processings that were in the training pipeline (such that the training runs faster). Our data preprocessing is also one of the reasons why our reproduced SMART-tiny (with 7M params, trained for 32 epochs) performs better than the original SMART-tiny. Our reproduced SMART-tiny ([test split submission](https://waymo.com/open/challenges/sim-agents/results/5ea7a3eb-7337/1737621512705000/), not published to the leaderboard but it's on test split) achieves an RMM of 0.7671.

**Comment by OrangeSodahub** [comment]
@zhejz Did you remember the time you spent on and the settings of data preprocessing? I found it seems too slow.

**Comment by zhejz** [comment]
It is slow, 2-3 hours for the val/test split and 1-2 days for the train split.

**Comment by OrangeSodahub** [comment]
@zhejz How about the corresponding settings? e.g. number of workers of multiprocessing. In my case, 2 days passesd but it only processed 20%

**Comment by zhejz** [comment]
Yes, you should change the `num_workers` [here](https://github.com/NVlabs/catk/blob/7fe4b803f237099b4e6cfa76c09d62b6686a2f03/src/data_preprocess.py#L516) according how many CPU cores you have. Try 16 or 32 for example.

**Comment by OrangeSodahub** [comment]
@zhejz Yeah I set exactly 16, but much slower than you

**Comment by OrangeSodahub** [comment]
Sorry I solved ...

**Comment by zhejz** [comment]
The node I used has 128 CPUs, and it takes 1-2 days. I guess your machine has maybe 32 CPUs? If that's the case then it makes sense.

**Comment by OrangeSodahub** [comment]
Another question is I want to know now I have the `pkl` files generated by SMART (only for map information, agent information is same as catK), can I directly use your training script? Not sure if there will be issues for map encodings

**Comment by zhejz** [comment]
Our map `pkl` [file](https://github.com/NVlabs/catk/blob/main/src/smart/tokens/map_traj_token5.pkl) is directly copied from the SMART repo, exactly the same as this [file](https://github.com/rainmaker22/SMART/blob/main/smart/tokens/map_traj_token5.pkl).

Our code is fully compatible in terms of token `pkl` files, whatever works for SMART, works also for us.

**Comment by OrangeSodahub** [comment]
Sounds great, I mean not token `pkl` files, but the preprocessed `scenario_id.pkl` files, where the map information come from SMART

**Comment by zhejz** [comment]
Hmmmm. I think it should be compatible, but some changes might be needed. For example we have more types of map polylines.

**Comment by zhejz** [comment]
The `val_open/acc` is around 0.810 after BC converges, and around 0.8092 after closed-loop fine-tuning converges.
The `val_open/loss` is around 2.276 after BC converges, and around 1.339 after closed-loop fine-tuning converges.
For BC, the loss is around 2.4 at the beginning, and drops to 2.3 after 10 epochs.

Yes the `train_mask` is different, there were many such kind of small changes we made.

**Comment by OrangeSodahub** [comment]
@zhejz Sorry to bother you again, did you use `num_workers=12` here?
https://github.com/NVlabs/catk/blob/7fe4b803f237099b4e6cfa76c09d62b6686a2f03/scripts/cache_womd.sh#L13
My node has 128 CPUs and I set `num_workers=32` however it cost ~2h to process 2k scenarios (~40k in total for val set).
